components/mailgun/actions/send-email/send-email.js

```diff
@@ -15,108 +19,118 @@ module.exports = {
     },
+    /* eslint-disable pipedream/default-value-required-for-optional-props */
     html: {
-      type: "string",
-      label: "Message Body (HTML)",
+      propDefinition: [
+        mailgun,
+        "body_html",
+      ],
       optional: true,
     },
+    /* eslint-enable pipedream/default-value-required-for-optional-props */
     testMode: {
       type: "boolean",
       label: "Send in test mode?",
       default: true,
-      description: "Enables sending in test mode. For more information, see the [Mailgun API documentation](https://documentation.mailgun.com/en/latest/api-sending.html#sending)",
+      description: "Enables sending in test mode. For more information, see the [Mailgun API " +
+        "documentation](https://documentation.mailgun.com/en/latest/api-sending.html#sending)",
     },
     dkim: {
       type: "boolean",
       label: "Use DKIM?",
       default: true,
-      description: "Enables or disables DKIM signatures. For more information, see the [Mailgun API documentation](https://documentation.mailgun.com/en/latest/api-sending.html#sending)",
+      description: "Enables or disables DKIM signatures. For more information, see the [Mailgun " +
+        "API documentation](https://documentation.mailgun.com/en/latest/api-sending.html#sending)",
       optional: true,
     },
     tracking: {
       type: "boolean",
       label: "Use Tracking?",
       default: true,
-      description: "Enables or disables tracking. For more information, see the [Mailgun API documentation](https://documentation.mailgun.com/en/latest/api-sending.html#sending)",
+      description: "Enables or disables tracking. For more information, see the [Mailgun API " +
+        "documentation](https://documentation.mailgun.com/en/latest/api-sending.html#sending)",
       optional: true,
     },
-    haltOnError: {
-      propDefinition: [
-        mailgun,
-        "haltOnError",
-      ],
-    },
+    ...props,
+  },
+  methods: {
+    ...methods,
   },
-  async run () {
-    try {
-      const msg = {
-        "from": `${this.fromName} <${this.from}>`,
-        "to": this.to,
-        "cc": this.cc,
-        "bcc": this.bcc,
-        "subject": this.subject,
-        "text": this.text,
-        "html": this.html,
-        "o:testmode": this.testMode,
-      };
-      if (this.replyTo) {
-        msg["h:Reply-To"] = this.replyTo;
-      }
-      if (this.dkim !== null) {
-        msg["o:dkim"] = this.dkim
-          ? "yes"
-          : "no";
-      }
-      if (this.tracking) {
-        msg["o:tracking"] = "yes";
-      }
-      return await this.mailgun.api("messages").create(this.domain, msg);
-    } catch (err) {
-      if (this.haltOnError) {
-        throw err;
-      }
-      return err;
+  async run() {
+    const msg = {
+      "from": `${this.fromName} <${this.from}>`,
+      "to": this.to,
+      "subject": this.subject,
+      "text": this.text,
+      "html": this.html,
+      "o:testmode": this.testMode,
+    };
```

Comment:
When I run this action with the `testMode` prop set to `true`, I get the error:
```
ErrorTypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer or Uint8Array. Received type boolean (true)
```

According to the [mailgun API docs](https://documentation.mailgun.com/en/latest/api-sending.html?highlight=testmode#sending), the `o:testmode` prop should be set to `yes` to enable sending in test mode.
Conditionally setting `msg["o:testmode"] = "yes"` fixed the error for me:

```suggestion
    };
    if (this.testMode) {
      msg["o:testmode"] = "yes";
    }
```